build/dart/rules.gni

```diff
@@ -5,7 +5,178 @@
 # This file has rules for making Dart packages and Dart-based Mojo applications.
 # The entrypoint is the dart_pkg rule.
 
+import("//build/compiled_action.gni")
 import("//build/module_args/dart.gni")
+import("//flutter/common/config.gni")
+import("//third_party/dart/build/dart/dart_action.gni")
+
+# Creates a dart kernel (dill) file suitalbe for use with gen_snapshot.
+#
+# Invoker must supply dart_main and package_config. Invoker may optionally
+# supply aot as a boolean and product as a boolean.
+template("dart_snapshot") {
+  assert(defined(invoker.main_dart), "main_dart is a required parameter.")
+  assert(defined(invoker.sources), "sources is a required parameter.")
```

Comment:
Why must the sources be explicitly specified even though the script action supports depfiles? Seems redundant. The `main_dart` and the package configuration ought to sufficient.

Reply:
Yeah this is wrong, removed it.